script.js

```diff
@@ -1,12 +1,547 @@
+  showSelect.addEventListener("change", handleShowSelectChange); 
+
+  addSelectOption(showSelect, "Select Show");
+
+  shows.sort((a, b) => a.name.localeCompare(b.name));
+
+  shows.forEach((show) => {
+    const option = document.createElement("option");
+    option.value = show.id;
+    option.textContent = show.name;
+    showSelect.appendChild(option);
+  });
+
+  showContainer.appendChild(showSelect);
+  searchSelectDiv.appendChild(showContainer);
+}
+
+function handleShowSelectChange() {
+  const selectedShowId = showSelect.value;
+  if (selectedShowId !== "select-show") {
+    getEpisodes(selectedShowId)
+      .then((episodes) => {
+        allEpisodes = episodes;
+        makePageForEpisodes(allEpisodes);
+
+        // Checking if the episode select field already exists
+        if (!episodeSelectContainer) {
+          // If it doesn't exist, add we add episode select field
+          episodeSelectField(allEpisodes);
+        } else {
+          // If it exists, we update its options with the new episodes
+          updateEpisodeSelect(allEpisodes);
+        }
+
+        // Checking if the search field already exists
+        if (!searchContainer) {
+          // If it doesn't exist,we  add the search field
+          addSearchField();
+        }
+
+        // Updating the search label text
+        const searchLabel = document.querySelector(".search-label");
+        searchLabel.textContent = `Displaying ${allEpisodes.length} / ${allEpisodes.length} episodes`;
+      })
+      .catch((error) => console.error(error));
+  } else {
+    showListing(allShows);
+    searchShows();
+
+    // Removing the episode select field and search field when we display shows
+    removeEpisodeSelectField();
+    removeSearchField();
+  }
+}
+
+// Updating the select field to get episodes from all shows
+function updateEpisodeSelect(episodes) {
+  const episodeSelect = document.querySelector(".episode-select");
+  if (episodes.length > 0) {
+    // Showing the episode select field
+    episodeSelectContainer.parentNode.style.display = "inline-flex";
+  } else {
+    // Hiding the episode select field
+    episodeSelectContainer.parentNode.style.display = "none";
+  }
+  episodeSelect.innerHTML = "";
+
+  const selectOption = document.createElement("option");
+  selectOption.value = "select-episode";
+  selectOption.textContent = "Select Episode";
+  episodeSelect.appendChild(selectOption);
+
+  episodes.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = episode.id;
+    option.textContent = `${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    episodeSelect.appendChild(option);
+  });
+}
+
+// Level 300 - Adding an episode selector
+function episodeSelectField(episodes) {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  episodeSelectContainer = document.createElement("div");
```

Comment:
when I press enter on the episode select it seems to refresh the page

Reply:
It just works with click until now I haven’t though about it, thank you for pointing out , I will search through to see how to make it work with enter